src/shared_utilities/analytic_time_varying_input.jl

```diff
@@ -1,15 +1,22 @@
 struct AnalyticTimeVaryingInput{F <: Function} <: AbstractTimeVaryingInput
-    # func here as to be GPU-compatible (e.g., splines are not)
+    # func here has to be GPU-compatible (e.g., splines are not)
```

Comment:
is it worth saying that it should also be performant? not allocating big amounts of memory, not looking up variables in global scope, etc

Reply:
Yes